src/interfaces/src/plugin/data/interfaces/interface/linked_list.h

```diff
@@ -0,0 +1,122 @@
+    }                                                                               \
+    while (0)
+
+#define INTERFACES_INTERFACE_LINKED_LIST_FREE(ll_ptr)                                                                                \
+        _Generic((ll_ptr),                                                                                                           \
+                 interfaces_interfaces_interface_ipv4_address_element_t**  : interfaces_interface_linked_list_ipv4_address_free      \
+                 interfaces_interfaces_interface_ipv4_neighbor_element_t** : interfaces_interface_linked_list_ipv4_neighbor_free     \
+                 interfaces_interfaces_interface_ipv6_address_element_t**  : interfaces_interface_linked_list_ipv6_address_free      \
+                 interfaces_interfaces_interface_ipv6_neighbor_element_t** : interfaces_interface_linked_list_ipv6_neighbor_free     \
+        )(ll_ptr)
+
+/* prepend since ordering doesn't matter - O(1) */
+#define INTERFACES_INTERFACE_LINKED_LIST_ADD_ELEMENT(ll_ptr, new_element_ptr)   \
+    do                                                                          \
+    {                                                                           \
+        LL_PREPEND(ll_ptr, new_element_ptr);                                    \
+    } while(0)                                                                  \
+
+#define INTERFACES_INTERFACE_LINKED_LIST_GET_ELEMENT_STRING(ll_ptr, element_ptr, member, value)   \
+    do                                                                                            \
+    {                                                                                             \
+        LL_FOREACH(ll_ptr, element_ptr) {                                                         \
+            if (strcmp(element_ptr->member, value) == 0) {                                        \
+                break;                                                                            \
+            }                                                                                     \
+        }                                                                                         \
+    } while(0)                                                                         
+
+#define INTERFACES_INTERFACE_LINKED_LIST_GET_ELEMENT_SCALAR(ll_ptr, element_ptr, member, value)   \
+    do                                                                                            \
+    {                                                                                             \
+        LL_SEARCH_SCALAR(ll_ptr, element_ptr, member, value);                                     \
+    } while(0)                                                                         
+
+void interfaces_interface_linked_list_ipv4_address_free(interfaces_interfaces_interface_ipv4_address_element_t **ll);
```

Comment:
The "linked_list" ipv4 prefix is not needed. Rather, the following syntax is preferred:
```c
void interfaces_interface_ipv4_address_list_free(...);
```
Adding the **list** ipv4_address sufix should keep the API consistent since the hash API uses the same approach. Same applies for the ipv6 equivalent and for both **address** and **neighbor** lists.